modules/core/src/main/java/io/fluo/core/impl/Environment.java

```diff
@@ -84,99 +105,48 @@ public Environment(Environment env) throws Exception {
     this.conn = env.conn;
     this.accumuloInstanceID = env.accumuloInstanceID;
     this.fluoInstanceID = env.fluoInstanceID;
-    this.observers = env.observers;
-    this.oraclePort = env.oraclePort;
     this.config = env.config;
     this.resources = new SharedResources(this);
-    this.rollbackTime = env.rollbackTime;
   }
 
   /**
-   * Constructs an environment from given FluoConfiguration
-   * @param configuration Configuration used to configure environment
+   * read configuration from zookeeper
+   * 
+   * @throws InterruptedException
+   * @throws KeeperException
    */
-  public Environment(FluoConfiguration configuration) {
-    this.config = configuration;
-    
+  private void readZookeeperConfig() {
+
```

Comment:
it could.. but then the curator would then need be closed by `Environment` rather than `SharedResources`